### PR TITLE
feature/#106: add and verify paas emails

### PR DESCRIPTION
**Pull request by radoslavzeman**
After discussion, we are not able to display all cards from all verified emails on one screen, so it's split into 2 screens ([emails (figma)](https://www.figma.com/file/3TppNabuUdnCChkHG9Vft7/paas-mpa?node-id=6291%3A7202&mode=dev) and then [cards for selected email (figma)](https://www.figma.com/file/3TppNabuUdnCChkHG9Vft7/paas-mpa?node-id=169%3A3185&mode=dev)).

- actual verification email is sent, but it's not completely ready yet, so I added temporary Verify button (token is returned from the first request for now).
- also implement `ContinueButton` with default "Continue" label, so we cam work with `useTranslation` hook easier (we can use keyPrefix also when Continue button is present easily)
- add props to parking cards components, implemented paren `ParkingCard` "wrapper" component
- implement Delete email action - it's not working on BE, but FE should be fine
- upgraded `expo-router` - there is fixed bug with `Stack.Screen` so we use it again for setting the title
  - use `Stack.Screen` where possible instead of `navigation` in useEffect

TODO
- handle pagination on cards and on emails
- split cards into Active and Expired
- no cards screen
- etc.

Note:
- types on BE are not correct e.g. for settings - they will be fixed on BE and regenerated on FE afterwards

### CODE REVIEW

**Review by lukybil — APPROVED** [review]
Very nice work! Just some small comments and the empty email bottom sheet which should be investigated.

File: components/parking-cards/ParkingCard.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  licencePlate: string | null | undefined
+  validUntil: string | null | undefined
+  cardNumber: string | null | undefined
+  remainingCredit: string | null | undefined
+}
+
+// TODO change card.name to zone
+const ParkingCard = ({ card }: Props) => {
+  switch (card.type) {
+    case 'ABONENT':
```
[the remark this answers is not in the document]
Comment (radoslavzeman) [reply components/parking-cards/ParkingCard.tsx:23]: Yes TS works nicely also with strings.

File: components/actions/ListRow.tsx [hunk 2]
```diff
@@ -5,19 +5,26 @@ import Icon, { IconName } from '@/components/shared/Icon'
...
 export type ListRowProps = {
-  icon: IconName
+  icon?: IconName
   label: string
   labelClassName?: string
 } & Omit<ViewProps, 'children'>
 
+/**
+ * ListRow should always be wrapped in PressableStyled (+ Link if needed)
+ *
```
[the remark this answers is not in the document]
Comment (radoslavzeman) [reply components/actions/ListRow.tsx:15]: So the thing is that you need to combine all props somehow. If you wrap it separately everywhere, you have control on what you pass to Pressable, what you pass to Link, and what to actual wrapped component.

We discussed it with Pinto shortly and agreed an this approach for now, but we ca discuss it further and find a better approach :)

File: app/parking-cards/index.tsx [hunk 3]
```diff
@@ -1,99 +1,70 @@
...
-        <Link asChild href="/parking-cards/enter-paas-account">
-          <IconButton name="add" accessibilityLabel={t('addParkingCards')} />
-        </Link>
-      ),
-    })
-  }, [navigation, t])
+  const { data, isPending, isError, error } = useQuery({
+    queryKey: ['VerifiedEmails'],
+    queryFn: () => clientApi.verifiedEmailsControllerVerifiedEmailsGetMany(1, 10),
+    select: (res) => res.data,
```
[the remark this answers is not in the document]
Comment (radoslavzeman) [reply app/parking-cards/index.tsx:22]: Yes, we can discuss what is the best approach for queryKeys, I agree to store them somehow.

File: app/parking-cards/[email].tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  const bottomSheetRef = useRef<BottomSheet>(null)
+
+  const layout = useWindowDimensions()
+
+  const [index, setIndex] = useState(0)
+  const [routes] = useState([
+    { key: 'active', title: t('activeCards') },
+    { key: 'expired', title: t('expiredCards') },
+  ])
```
[the remark this answers is not in the document]
Comment (radoslavzeman) [reply app/parking-cards/[email].tsx:86]: This is just copied from docs actually. But also translation is needed for titles. Idk, I would keep it like this if it does not causes problems.

**Pull request merged**